js/app.js

```diff
@@ -1,89 +1,120 @@
-document.addEventListener(`DOMContentLoaded`, () =>{
+const registrar = document.querySelector(`#registrar`);
+const addInviteeInput = registrar.querySelector(`#add_invitee_input`);
+const invitedUl = document.querySelector(`#invitedList`);
+const hideNotRespondedCheckbox = document.querySelector(`.main input`);
+const emptyInviteeTile = document.querySelector('#empty_invitee').firstElementChild;
 
-	const form = document.querySelector(`#registrar`);
-	const formInput = form.querySelector(`input`);
-	const invitedUl = document.querySelector(`#invitedList`);
-	const hideNotResponded = document.querySelector(`.main input`);
+// added one user add beggining for testing purpose
+const Czarek = new Invitee(0, 'Czarek', true);
 
-	function newElement(elementParent, element) {
-		const name = document.createElement(element);
-		elementParent.appendChild(name)
-		return name
-	}
+let invitees = [Czarek];
+let nextId = invitees.length;
 
-	function text(elementParent, element, text) {
-		const name = newElement(elementParent, element);
-		name.textContent = text;
-		return name;
-	}
+/* -----------------
+ Object Functions
+------------------ */
 
-	function type(elementParent, element, type) {
-		const name = newElement(elementParent, element);
-		name.type = type;
-		return name;
-	}
+function addNewInvitee() {
+	const newInvitee = new Invitee(nextId, addInviteeInput.value);
+	invitees = [...invitees, newInvitee];
+	addInviteeInput.value = ``;
+	nextId++
+}
 
-	function submitForm() {
-		const newLi = newElement(invitedUl, 'li');
-		const newSpan = text(newLi, `span`, formInput.value);
-		const newInput = type(newLi, `input`, 'text');
-		newInput.value = formInput.value;
-		newInput.classList.add(`invisible`);
-		const label = text(newLi, `label`, `Confirmed`);
-		label.setAttribute(`for`, `confirmed`);
-		const checkbox = type(label, 'input', `checkbox`);
-		checkbox.setAttribute(`name`, `confirmed`);
-		const edit = text(newLi, 'button', 'Edit');
-		const remove = text(newLi, 'button', `Remove`);
-	}
+function indexOfId(id) {
+	const index = invitees.findIndex(invitee => invitee.id === id);
+	return index;
+}
 
+function editOrRemoveInvitee(button) {
+	const buttonName = button.textContent.toLowerCase();
+	const inviteeId = parseInt(button.dataset[buttonName]);
+	const inviteeIndex = indexOfId(inviteeId);
+	const invitee = invitedUl.querySelector(`#invitee-${inviteeId}`);
+	const buttonActions = {
```

Comment:
Moim zdaniem to mógłby być osobny obiekt, poza scopem tej funkcji, wtedy button actions byłyby bardziej reużywalne. Trochę mam też wrażenie, że ulokowanie tego obiektu tutaj wpływa źle na czytelność - ale z tym już można polemizować